cmd/node/node.go

```diff
@@ -468,40 +469,67 @@ func (app SpacemeshApp) stopServices() {
 
 }
 
-func getEdIdentity() (*signing.EdSigner, error) {
-	dataDir, err := filesystem.GetSpacemeshDataDirectoryPath()
+func (app *SpacemeshApp) getEdIdentity() (*signing.EdSigner, error) {
+	f, err := app.getIdentityFile()
 	if err != nil {
-		log.Error("Could not get data path err=%v", err)
-		return nil, err
-	}
+		log.Warning("Failed to find identity file: %v", err)
 
-	f := dataDir + "/" + identityFile
-	buff, err := ioutil.ReadFile(f)
-	if os.IsNotExist(err) {
 		edSgn := signing.NewEdSigner()
-		log.Warning("Identity file not found. Public key of new identity is %v", edSgn.PublicKey())
-		err := ioutil.WriteFile(f, edSgn.ToBuffer(), 0644)
+		f = filepath.Join(app.Config.POST.DataDir, edSgn.PublicKey().String(), identityFileName)
+		err := os.MkdirAll(filepath.Dir(f), filesystem.OwnerReadWriteExec)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create directory for identity file: %v", err)
+		}
+		err = ioutil.WriteFile(f, edSgn.ToBuffer(), filesystem.OwnerReadWrite)
```

Comment:
I saved the private key in this folder merely as a tool for the init factory. Reusing it for persisting the node identity might be fine, because the folder already supports multi identities. But we might want to consider symmetric encryption on disk.

Reply:
Encryption with what key? Where would you store it?

Reply:
It won't be stored. It will be the user password he need to remember for unlocking the node.

Reply:
I'm not sure if I'm for/against this. The main downside is that the node won't be able to start without user involvement, e.g. if there's a power outage and the computer restarts.

There's definitely a tradeoff here and we may want to make encryption an option. However, it's definitely not in the scope of this PR.

@zalmen WDYT?

Reply:
Locking need to be introduced anyway on the wallet side. So we can consider to combine these two operations.

Reply:
Locking need to be introduced anyway on the wallet side. Might be possible to reuse that on the node identity.